addons/mail/static/tests/chat_window/chat_window_tests.js

```diff
@@ -404,6 +404,25 @@ QUnit.test("Close active thread action in chatwindow on ESCAPE", async () => {
     await contains(".o-mail-ChatWindow");
 });
 
+QUnit.test("Exit chatwindow thread rename on ESCAPE without closing chat window", async () => {
+    const pyEnv = await startServer();
+    pyEnv["discuss.channel"].create({
+        name: "General",
+        channel_member_ids: [
+            Command.create({ partner_id: pyEnv.currentPartnerId, is_minimized: true }),
+        ],
+        create_uid: pyEnv.currentUserId,
+    });
+    await start();
+    await contains(".o-mail-ChatWindow");
+    await click(".o-mail-ChatWindow-command", { text: "General" });
+    await click(".dropdown-item", { text: "Rename" });
+    await contains(".o-mail-AutoresizeInput:focus");
+    triggerHotkey("Escape");
+    await contains(".o-mail-AutoresizeInput:focus", { count: 0 });
+    await contains(".o-mail-ChatWindow");
```

Comment:
Note that `:focus` is not a DOM mutation, so if the input is not focused initially then this will have to wait for `contains()` timeout after 3 seconds.

We could either ping every x times like web `contains`, or make mail `contains` more sophisticated with detecting `:focus` and putting some code in place to detect them in real time... Or you can put an artificial mutation that matches it, e.g. with `.o-focused` classname.

I would suggest to do the later, this is the code that is the most straight-forward to do for now.

Reply:
But doing so would require to add a new state variable `isFocused`. Which in my opinion, isn't justified just to make the test case run a bit faster.

Reply:
@rapr-odoo It's this or the test can fail non-deterministically (timeout of`contains(:focus)` x 2)... Or we improve `contains()` so it detects `:focus` in selector to observe such change immediately, but that's much more complex than just have `isFocused` state.